### PR TITLE
Added fixtures and temporary dirs for tests.

### LINKED ISSUE #88: Make the tests use a temporary directory and a fixture

### PR DESCRIPTION
**Pull request by melissawm**
Including some configs to make fixtures more reusable.

Fixes #88

### CODE REVIEW

File: versioned_hdf5/wrappers.py [hunk 1]
```diff
@@ -296,6 +296,7 @@ def _make_new_dset(shape=None, dtype=None, data=None, chunks=None,
                   track_order=None, dcpl=None):
     """ Return a new low-level dataset identifier """
 
+    print(f"shape = {shape}")
```
Comment (asmeurer) [inline versioned_hdf5/wrappers.py:299]: I think this shouldn't be here.

File: versioned_hdf5/tests/test_api.py [hunk 2]
```diff
@@ -10,1037 +10,925 @@
...
-        group = file['resize_after_create_smaller']
         assert group['data'].shape == (DEFAULT_CHUNK_SIZE - 4,)
         assert_equal(group['data'][:], 1.0)
 
-        file.close()
+    group = vfile['resize_after_create_smaller']
+    assert group['data'].shape == (DEFAULT_CHUNK_SIZE - 4,)
+    assert_equal(group['data'][:], 1.0)
 
-def test_resize_unaligned():
```
Comment (asmeurer) [inline versioned_hdf5/tests/test_api.py:564]: This test was deleted.

**Comment by asmeurer** [comment]
Do we still need the `setup()` helper function? I see it's still used in some places.

**Comment by melissawm** [comment]
> Do we still need the `setup()` helper function? I see it's still used in some places.

The difficulty is that some tests re-run the setup routine several times, and this can't be done with a fixture, which is run once before the test. I couldn't figure a way to remove the setup function from those.

**Comment by scopatz** [comment]
@melissawm - is this ready for review?

**Pull request merged**